lib/oregon_digital/uri_enabled_string_field.rb

```diff
@@ -0,0 +1,49 @@
+module OregonDigital
+  # Simple hackery to make a field that has a URI auto-switch to read-only
+  # while still being able to be removed.  Called from a simpleform input to
+  # generate HTML for the view.
+  class URIEnabledStringField
+    def initialize(builder, attribute_name, options)
```

Comment:
```ruby
pattr_initialize :builder, :attribute_name, :options
```
will do this and give you private accessors for builder/attribute_name/options so you don't have to use the instance variables.